tests/snippets/global_nonlocal.py

```diff
@@ -21,3 +22,37 @@ def y():
 
 res = x()
 assert res == 3, str(res)
+
+# Invalid syntax:
+src = """
+b = 2
+global b
+"""
+
+try:
+    exec(src)
+except Exception as ex:
+    # print(ex)
+    pass
+else:
+    raise RuntimeError('Must raise')
+
+# Invalid syntax:
```

Comment:
Are we ready to test invalid nonlocal in a function too?

Reply:
I think so, what is the wrong case here?

Reply:
I was thinking simply:
```
def f():
    nonlocal x
    return x
```

Reply:
Yes, this is checked. This will raise a SyntaxError.